File: e-learning.web/src/@types/auditLog.ts

export type AuditLogAction = 'login' | 'logout' | 'view' | 'download' | 'create' | 'update' | 'delete' | 'other';

export type AuditLogEntry = {
  id: number;
  occurredAt: string;
  action: string;
  entityId: string;
  entityType: string;
  metadata: string; // JSON string
  schoolId: string | null;
  tenantId: string;
  userId: string;
  // UI helper fields (populated during mapping)
  ipAddress?: string;
  username?: string;
};

export type AuditLogFilter = {
  startDate: string;
  endDate: string;
  userId: string;
  tenantId: string;
  schoolId: string;
  action: string;
};

File: e-learning.backend/src/Modules/AuditLogs/Aig.Lms.Modules.AuditLogs.Domain/AuditLog.cs
namespace Aig.Lms.Modules.AuditLogs.Domain;

public sealed class AuditLog
{
    public long Id { get; init; }
    public DateTime OccurredAt { get; init; }
    public Guid? TenantId { get; init; }
    public Guid? SchoolId { get; init; }
    public Guid? UserId { get; init; }
    public string Action { get; init; } = string.Empty;
    public string EntityType { get; init; } = string.Empty;
    public Guid? EntityId { get; init; }
    public string? IpAddress { get; init; }
    public string? UserAgent { get; init; }
    public string? Metadata { get; init; }
}
